### PR TITLE
Tdl 19964 add missing tap tester tests

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change
 - Updated several test cases according to the code changes
- Skipped missing fields from the all_fields test case
- Added assertion for primary keys in the automatic_fields test
- Added an assertion that replication value is greater than the bookmark in bookmarks test case
- Added an assertion for no duplicate metadata entry in the discover test case
- Added an assertion for actual replication that matches our expected replication method in the discover test case
- Added pagination test case
- Added replication key is greater or equal to start_date in start_date test
- Added primary keys assertion in start_date test

Note: Test cases will pass once after all the code of dict-to-class will be merged

# Manual QA steps
- 

# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by savan-chovatiya — CHANGES_REQUESTED** [review]

File: tests/test_typeform_bookmarks.py [hunk 1]
```diff
@@ -107,7 +58,7 @@ def test_run(self):
         ##########################################################################
 
         new_states = {'bookmarks': dict()}
-        simulated_states = self.calculated_states_by_stream(first_sync_bookmarks)
+        simulated_states = {'forms': {'last_updated_at': '2022-07-22T16:00:47Z'}, 'landings': {'mVn2wE55': {'landed_at': '2022-07-21T00:00:00Z'}, 'xJ8emTTy': {'landed_at': '2022-07-24T16:49:54Z'}}, 'answers': {'mVn2wE55': {'landed_at': '2022-07-21T00:00:00Z'}, 'xJ8emTTy': {'landed_at': '2022-07-24T16:49:54Z'}}}
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/test_typeform_bookmarks.py]: Added multiline for better readability

File: tests/test_typeform_bookmarks.py [hunk 2]
```diff
@@ -183,30 +133,36 @@ def test_run(self):
...
-                                    msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
-                                )
-
-
-                            # Verify the number of records in the 2nd sync is less then the first
+                            if stream != 'answers':
+                                for record in second_sync_messages:
+
+                                    # Verify the second sync records respect the previous (simulated) bookmark value
+                                    # replication_key_value = [record.get(replication_key) if record.get('_sdc_form_id') == form_key]
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/test_typeform_bookmarks.py]: removed commented code

File: tests/test_typeform_pagination.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+#for tap-typeform pagination
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/test_typeform_pagination.py]: removed unwanted comments

File: tests/test_typeform_start_date.py [hunk 4]
```diff
@@ -1,42 +1,55 @@
...
-
-        self.start_date_1 = self.get_properties().get("start_date")
-        self.start_date_2 = self.timedelta_formatted(self.start_date_1, days=3)
-
+        """
+        Test that the start_date configuration is respected
+        • verify that a sync with a later start date has at least one record synced
+        and less records than the 1st sync with a previous start date
+        • verify that each stream has less records than the earlier start date sync
+        • verify all data from later start data has bookmark values >= start_date
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/test_typeform_start_date.py]: Updated to uppercase

File: tests/test_typeform_bookmarks.py [hunk 5]
```diff
@@ -183,30 +133,36 @@ def test_run(self):
...
+                                for record in first_sync_messages:
+                                    # Verify the first sync bookmark value is the max replication key value for a given stream
+                                    if record.get('_sdc_form_id') == form_key:
+                                        replication_key_value = record.get(replication_key)
+                                        self.assertLessEqual(
+                                            replication_key_value, first_bookmark_value,
+                                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                                        )
+                                    else:
+                                        continue
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/test_typeform_bookmarks.py]: removed the else

File: tests/test_typeform_bookmarks.py [hunk 6]
```diff
@@ -183,30 +133,36 @@ def test_run(self):
...
-                                # Verify the first sync bookmark value is the max replication key value for a given stream
-                                replication_key_value = record.get(replication_key)
-                                self.assertLessEqual(
-                                    replication_key_value, first_bookmark_value_utc,
-                                    msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
-                                )
-
-
-                            # Verify the number of records in the 2nd sync is less then the first
+                            if stream != 'answers':
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/test_typeform_bookmarks.py:141]: Added comments

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**